app/src/verifier.h

```diff
@@ -362,17 +363,106 @@ namespace scitt::verifier
         }
         else if (phdr.issuer.has_value())
         {
-          // IETF SCITT claim
-          key = process_ietf_profile(
-            phdr, tx, current_time, resolution_cache_expiry, configuration);
-
-          try
+          if (phdr.issuer->starts_with("did:x509"))
           {
-            cose::verify(data, key);
+            // IETF SCITT did:x509 claim
+            if (phdr.x5chain.has_value())
+            {
+              // NB: In later revisions of SCITT, x5chain is unprotected, and
+              // only x5t is. This logic will need to authenticate x5chain[0]
+              // against x5t before it can proceed to verify the signature.
+              // Verify the signature as early as possible
+              OpenSSL::Unique_X509 leaf =
```

Comment:
Why is the existing approach not being used: validate chain against the trust store and then use the pub key like in the notary profile processing above, e.g.:

```
auto roots = x509_root_store(tx);
auto cert = verify_chain(roots, x5chain);
key = PublicKey(cert, std::nullopt);
```

Reply:
Because `verify_chain()` is duplicating what `didx509::resolve()` does, but without being apply to apply the query along the way, so we would be checking all the signatures twice effectively.